File: src/main/java/cloud/common/EPFD.java
package cloud.common;

import cloud.epfd.HealthCheckerInit;
import se.sics.kompics.PortType;

/**
 * 
 * @author Amir Moulavi
 * @date 2011-03-27
 *
 */

public class EPFD extends PortType {
	{
		positive(Suspect.class);
		positive(Restore.class);
		negative(HealthCheckerInit.class);
		negative(ConsiderInstance.class);
		negative(InstanceKilled.class);
	}
}

File: src/main/java/cloud/common/EPFD.scala
package cloud.common

import se.sics.kompics.PortType
import cloud.epfd.HealthCheckerInit

/**
 * @author Amir Moulavi
 * @date 2011-07-09
 *
 */
class EPFD extends PortType {

  positive(classOf[Suspect])
  positive(classOf[Restore])
  negative(classOf[HealthCheckerInit])
  negative(classOf[ConsiderInstance])
  negative(classOf[InstanceKilled])

}